internal/service/alarm_notification_handler.go

```diff
@@ -0,0 +1,287 @@
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+	selectorEvaluator, err := search.NewSelectorEvaluator().
+		SetLogger(b.logger).
+		SetPathEvaluator(pathEvaluator.Evaluate).
+		Build()
+	if err != nil {
+		return
+	}
+	// Create the jq tool:
+	jqTool, err := jq.NewTool().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+
+	// Check that extensions are at least syntactically valid:
+	for _, extension := range b.extensions {
+		_, err = jqTool.Compile(extension)
+		if err != nil {
+			return
+		}
+	}
+
+	alarmSubscriptionSearcher := newAlarmSubscriptionSearcher()
+	alarmSubscriptionSearcher.SetLogger(b.logger).SetJqTool(jqTool).build()
+
+	// create persist storeage option
+	persistStore := persiststorage.NewKubeConfigMapStore().
+		SetNameSpace(TestNamespace).
+		SetName(TestConfigmapName).
+		SetFieldOwnder(FieldOwner).
+		SetJsonAPI(&jsonAPI).
+		SetClient(b.kubeClient)
+
+	// http client to send out notification
+	// use 2 sec first
+	httpClient := http.Client{Timeout: 2 * time.Second}
+
+	if singleAlarmNotificationHandle != nil {
+		panic("There is an existing handler instance")
+	}
+	// Create and populate the object:
+	result = &alarmNotificationHandler{
+		logger:                    b.logger,
+		loggingWrapper:            b.loggingWrapper,
+		cloudID:                   b.cloudID,
+		extensions:                slices.Clone(b.extensions),
+		selectorEvaluator:         selectorEvaluator,
+		jsonAPI:                   jsonAPI,
+		jqTool:                    jqTool,
+		subscriptionMapMemoryLock: &sync.RWMutex{},
+		subscriptionMap:           &map[string]data.Object{},
+		persistStore:              persistStore,
+		subscriptionSearcher:      alarmSubscriptionSearcher,
+		httpClient:                httpClient,
+	}
+	if singleAlarmNotificationHandle != nil {
+		panic("There is an existing handler instance")
+	}
+	singleAlarmNotificationHandle = result
+
+	b.logger.Debug(
+		"alarmNotificationHandler build:",
+		"CloudID", b.cloudID,
+	)
+
+	err = result.recoveryFromPersistStore(ctx)
+	if err != nil {
+		b.logger.Error(
+			"alarmNotificationHandler failed to recovery from persistStore ",
+			slog.String(": ", err.Error()),
+		)
+	}
+
+	err = result.watchPersistStore(ctx)
+	if err != nil {
+		b.logger.Error(
+			"alarmNotificationHandler failed to watch persist store changes ",
+			slog.String(": ", err.Error()),
+		)
+	}
+	return
+}
+
+func (h *alarmNotificationHandler) recoveryFromPersistStore(ctx context.Context) (err error) {
+	newMap, err := persiststorage.GetAll(h.persistStore, ctx)
+	if err != nil {
+		return
+	}
+	err = h.assignSubscriptionMap(&newMap)
+	if err != nil {
+		h.logger.Error(
+			"alarmNotificationHandler failed building the indexes ",
+			slog.String(": ", err.Error()),
+		)
+
+	}
+	return
+}
+
+func (h *alarmNotificationHandler) watchPersistStore(ctx context.Context) (err error) {
+	err = persiststorage.ProcessChangesWithFunction(h.persistStore, ctx, ProcessStorageChanges)
```

Comment:
`ProcessStorageChanges` should be a private method of the handler, something like this:

```go
func (h *alarmNotificationHandler) processStorageChanges(...) {
        ....
}
```

Then here you can pass that method as parameter:

```go
err = persiststorage.ProcessChangesWithFunction(h.persistStore, ctx, h.processStorageChanges)
```